### ISSUE TITLE
Question about  OmniRL project

### ISSUE THREAD

**Issue by SharingChen** [airs-cuhk/airsoul#22]
我尝试用开源的配置文件和训练好的模型参数文件 运行generate.py, 但是无法得到预期的结果.
配置文件: config_16-128state_stage2_gsa_l32k_t512k_hot.yaml
模型参数文件: model.pth

Q1: 配置文件名为 stage2   是否模型参数也是stage2 训练后的结果,而不是stage3.
Q2: 配置文件中关于action_decode的配置,最终输出维度是否应该是5. 而且该配置文件在使用中缺少一些关键参数配置,例如'agent_num'.
```
  action_decode:
      output_type: Discrete
      input_size: 512
      hidden_size:
          - 512
          - 6
      layer_norm: True
      residual_connect: True
      dropout: 0.0
      is_frozen: False
```
能否更新配置文件和训练好的模型参数, 能够直接运行generate.py文件,以'lake4x4'环境配置为例,可以得到文章中的预期实验结果,感谢!

**Comment by WorldEditors** [comment]
感谢提问。可否将你运行的终端日志贴上来看看？

**Comment by SharingChen** [comment]
这是我的运行日志

> 20250326-14:57:55[DEBUG] Use Parallel GPUs: 8
> Final configuration:
> 
> Configure
> 
> run_name = gsa_20241216_GSA_16-128state_Stage2
> master_port = 10666
> load_model_path = /disk1/***/RL4AntiJamming_Exp/code/RL-antiJammer/anymdp_Omnirl/model/
> log_config:
>         use_tensorboard = True
>         tensorboard_log = /disk1/***/RL4AntiJamming_Exp/code/RL-antiJammer/anymdp_Omnirl/log/runs
>         training_log = /disk1/***/RL4AntiJamming_Exp/code/RL-antiJammer/anymdp_Omnirl/log/train.log
>         evaluation_log = /disk1/***/RL4AntiJamming_Exp/code/RL-antiJammer/anymdp_Omnirl/log/eval.log
> model_config:
>         context_warmup = 4000
>         max_position_loss_weighting = 128000
>         action_dim = 5
>         policy_loss_type = CrossEntropy
>         rsa_type = potar
>         state_encode:
>                 output_type = Discrete
>                 input_type = Discrete
>                 input_size = 128
>                 hidden_size = 512
>                 dropout = 0.0
>                 is_frozen = False
>         action_encode:
>                 output_type = Discrete
>                 input_type = Discrete
>                 input_size = 5
>                 hidden_size = 512
>                 dropout = 0.0
>                 is_frozen = False
>         prompt_encode:
>                 output_type = Discrete
>                 input_type = Discrete
>                 input_size = 8
>                 hidden_size = 512
>                 dropout = 0.0
>                 is_frozen = False
>         tag_encode:
>                 output_type = Discrete
>                 input_type = Discrete
>                 input_size = 8
>                 hidden_size = 512
>                 dropout = 0.0
>                 is_frozen = False
>         reward_encode:
>                 output_type = Continuous
>                 input_type = Continuous
>                 input_size = 1
>                 hidden_size = 512
>                 dropout = 0.0
>                 is_frozen = False
>         state_decode:
>                 output_type = Discrete
>                 input_size = 512
>                 hidden_size = [128]
>                 layer_norm = True
>                 residual_connect = False
>                 dropout = 0.0
>                 is_frozen = False
>         action_decode:
>                 output_type = Discrete
>                 input_size = 512
>                 hidden_size = [512, 6]
>                 layer_norm = True
>                 residual_connect = True
>                 dropout = 0.0
>                 is_frozen = False
>         reward_decode:
>                 output_type = Continuous
>                 input_size = 512
>                 hidden_size = [1]
>                 layer_norm = True
>                 residual_connect = False
>                 dropout = 0.0
>                 is_frozen = False
>         action_diffusion:
>                 enable = False
>                 T = 20
>                 hidden_size = 512
>                 condition_size = 512
>                 inner_hidden_size = 2048
>                 beta = [0.05, 0.2]
>                 dropout = 0.0
>                 is_frozen = False
>         state_diffusion:
>                 enable = False
>                 T = 20
>                 hidden_size = 512
>                 condition_size = 512
>                 inner_hidden_size = 2048
>                 beta = [0.05, 0.2]
>                 dropout = 0.0
>                 is_frozen = False
>         causal_block_bak_1:
>                 model_type = TRANSFORMER
>                 num_layers = 18
>                 hidden_size = 512
>                 nhead = 8
>                 inner_hidden_size = 512
>                 dropout = 0.1
>                 context_window = -1
>                 checkpoints_density = -1
>                 position_encoding_size = 12000
>                 use_layer_norm = True
>                 use_blockrecurrence = True
>                 memory_length = 3000
>                 memory_type = KV
>         causal_block:
>                 model_type = GSA
>                 num_layers = 18
>                 hidden_size = 512
>                 inner_hidden_size = 1024
>                 dropout = 0.1
>                 nhead = 4
>                 memory_length = 64
>                 position_encoding_size = 12000
>                 use_layer_norm = True
>                 use_blockrecurrence = True
>                 checkpoints_density = -1
>                 gate_bound = 22.0
>                 memory_type = MEM
>                 is_frozen = False
>                 is_generate = True
>         causal_block_bak_3:
>                 model_type = RWKV6
>                 num_layers = 12
>                 hidden_size = 512
>                 inner_hidden_size = 1024
>                 dropout = 0.1
>                 nhead = 16
>                 expand_k = 1
>                 expand_v = 2
>                 hidden_ratio = 3.5
>                 position_encoding_size = 12000
>                 use_layer_norm = True
>                 use_blockrecurrence = True
>                 checkpoints_density = -1
>                 memory_length = 0
>                 memory_type = MEM
>         causal_block_bak_4:
>                 model_type = Mamba
>                 num_layers = 12
>                 hidden_size = 512
>                 inner_hidden_size = 1024
>                 expand = 2
>                 d_conv = 4
>                 d_state = 16
>                 dropout = 0.1
>                 position_encoding_size = 12000
>                 use_layer_norm = True
>                 use_blockrecurrence = True
>                 checkpoints_density = -1
>                 memory_length = 0
>                 memory_type = MEM
> train_config:
>         max_epochs = 20
>         batch_size = 4
>         seq_len = 32000
>         seg_len = 4000
>         lr = 0.0002
>         lr_decay_interval = 1000
>         lr_start_step = 16000
>         data_path = /data/dataset/anymdp_12_10/UniTask_s128_a5_t512k_l32k/
>         save_model_path = /data/shaopt/checkpoints/20250127_GSA_16-128state_l32k_t512k/ckpts/
>         max_save_iterations = 1000
>         state_dropout = 0.0
>         reward_dropout = 0.0
>         lossweight_policymodel = 0.8
>         lossweight_worldmodel_states = 0.1
>         lossweight_worldmodel_rewards = 0.1
>         lossweight_entropy = 1e-12
>         lossweight_l2 = 1e-12
>         use_amp = False
>         use_scaler = False
> test_config:
>         batch_size = 8
>         data_path = /data/dataset/anymdp_12_10/UniTask_s128_a5_t256_l64k_d256
>         output = /data/shaopt/checkpoints/20250127_GSA_16-128state_l32k_t512k/offline_eval/
>         seq_len = 64000
>         seg_len = 8000
> generator_config:
>         agent_num = 1
>         env = lake4x4
>         map_env_discrete = True
>         action_clip = 4
>         skip_frame = 0
>         epoch_numbers = 3
>         downsample_trail = 10
>         decoding_strategy:
>                 T_ini = 1
>                 T_fin = 0.1
>                 T_step = 250
>                 decay_type = Linear
>         max_trails = 500
>         max_steps = 200
>         max_total_steps = 0
>         learn_from_data = False
>         data_root = /disk1/***/RL4AntiJamming_Exp/code/RL-antiJammer/OmniRL/airsoul-main/projects/OmniRL/gym_data/data
>         run_icl = True
>         use_dym_tag = False
>         run_benchmark:
>                 run_opt = False
>                 run_online = False
>                 run_random = False
>         benchmark_model_save_path = /disk1/***/RL4AntiJamming_Exp/code/RL-antiJammer/OmniRL/airsoul-main/projects/OmniRL/gym_data
>         output = ./online_eval/
>         save_gif = False
>         save_gif_gap = 50
> 
> 
> 
> 20250326-14:58:01[DEBUG] Main gpu 0 cuda:0
> 20250326-14:58:02[DEBUG] RSA Decision Model initialized, total params: 40870663
> 20250326-14:58:02[DEBUG] Causal Block Parameters: 40192000
> 20250326-14:58:06[WARNING] Current parameters module.mask_query not found in the checkpoint Skipping loading...
> 20250326-14:58:07[DEBUG] -------------------- Load model success --------------------
> Load model 0 from /disk1/***/RL4AntiJamming_Exp/code/RL-antiJammer/anymdp_Omnirl/model/ for GPU 0
> 20250326-14:58:07[DEBUG] GPU 0 start processing epoch 0 ...
> Run ICL Only.
> Load model 0 from /disk1/***/RL4AntiJamming_Exp/code/RL-antiJammer/anymdp_Omnirl/model/ for GPU 2
> 20250326-14:58:09[DEBUG] GPU 2 start processing epoch 0 ...
> Run ICL Only.
> Load model 0 from /disk1/***/RL4AntiJamming_Exp/code/RL-antiJammer/anymdp_Omnirl/model/ for GPU 7
> 20250326-14:58:09[DEBUG] GPU 7 start processing epoch 0 ...
> Run ICL Only.
> Load model 0 from /disk1/***/RL4AntiJamming_Exp/code/RL-antiJammer/anymdp_Omnirl/model/ for GPU 5
> 20250326-14:58:09[DEBUG] GPU 5 start processing epoch 0 ...
> Run ICL Only.
> Load model 0 from /disk1/***/RL4AntiJamming_Exp/code/RL-antiJammer/anymdp_Omnirl/model/ for GPU 3
> 20250326-14:58:09[DEBUG] GPU 3 start processing epoch 0 ...
> Run ICL Only.
> Load model 0 from /disk1/***/RL4AntiJamming_Exp/code/RL-antiJammer/anymdp_Omnirl/model/ for GPU 1
> 20250326-14:58:09[DEBUG] GPU 1 start processing epoch 0 ...
> Run ICL Only.
> Load model 0 from /disk1/***/RL4AntiJamming_Exp/code/RL-antiJammer/anymdp_Omnirl/model/ for GPU 6
> 20250326-14:58:09[DEBUG] GPU 6 start processing epoch 0 ...
> Run ICL Only.
> Load model 0 from /disk1/***/RL4AntiJamming_Exp/code/RL-antiJammer/anymdp_Omnirl/model/ for GPU 4
> 20250326-14:58:09[DEBUG] GPU 4 start processing epoch 0 ...
> Run ICL Only.
> 20250326-14:58:20 trail_idx:001 total_steps:002 step:002 reward:0.000e+00 state_prediction:5.421 reward_prediction:0.890 success_rate:0.000e+00
> 20250326-14:58:22 trail_idx:002 total_steps:010 step:008 reward:0.000e+00 state_prediction:3.713 reward_prediction:0.206 success_rate:0.000e+00
> 20250326-14:58:23 trail_idx:003 total_steps:020 step:010 reward:0.000e+00 state_prediction:1.877 reward_prediction:0.128 success_rate:0.000e+00
> 20250326-14:58:24 trail_idx:004 total_steps:024 step:004 reward:0.000e+00 state_prediction:1.174 reward_prediction:0.110 success_rate:0.000e+00
> 20250326-14:58:27 trail_idx:005 total_steps:045 step:021 reward:0.000e+00 state_prediction:1.614 reward_prediction:1.682e-02 success_rate:0.000e+00
> 20250326-14:58:28 trail_idx:006 total_steps:050 step:005 reward:0.000e+00 state_prediction:2.574 reward_prediction:0.210 success_rate:0.000e+00
> 20250326-14:58:29 trail_idx:007 total_steps:056 step:006 reward:0.000e+00 state_prediction:0.557 reward_prediction:2.338e-02 success_rate:0.000e+00
> 20250326-14:58:29 trail_idx:008 total_steps:058 step:002 reward:0.000e+00 state_prediction:2.712 reward_prediction:0.299 success_rate:0.000e+00
> 20250326-14:58:30 trail_idx:009 total_steps:065 step:007 reward:0.000e+00 state_prediction:4.526 reward_prediction:0.140 success_rate:0.000e+00
> 20250326-14:58:31 trail_idx:010 total_steps:072 step:007 reward:0.000e+00 state_prediction:0.834 reward_prediction:1.560e-02 success_rate:0.000e+00
> 20250326-14:58:33 trail_idx:011 total_steps:086 step:014 reward:0.000e+00 state_prediction:0.749 reward_prediction:2.770e-02 success_rate:0.000e+00
> 20250326-14:58:38 trail_idx:012 total_steps:122 step:036 reward:0.000e+00 state_prediction:0.776 reward_prediction:2.676e-02 success_rate:0.000e+00
> 20250326-14:58:39 trail_idx:013 total_steps:126 step:004 reward:0.000e+00 state_prediction:1.203 reward_prediction:1.461e-02 success_rate:0.000e+00
> 20250326-14:58:40 trail_idx:014 total_steps:132 step:006 reward:0.000e+00 state_prediction:0.641 reward_prediction:3.734e-04 success_rate:0.000e+00
> 20250326-14:58:45 trail_idx:015 total_steps:171 step:039 reward:0.000e+00 state_prediction:0.280 reward_prediction:1.865e-02 success_rate:0.000e+00
> 20250326-14:58:48 trail_idx:016 total_steps:194 step:023 reward:0.000e+00 state_prediction:0.778 reward_prediction:4.289e-02 success_rate:0.000e+00
> 20250326-14:58:51 trail_idx:017 total_steps:210 step:016 reward:1.000 state_prediction:0.966 reward_prediction:7.233e-02 success_rate:0.100
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> 20250326-14:59:20 trail_idx:018 total_steps:410 step:200 reward:0.000e+00 state_prediction:3.500e-02 reward_prediction:3.587e-04 success_rate:9.000e-02
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> 20250326-14:59:49 trail_idx:019 total_steps:610 step:200 reward:0.000e+00 state_prediction:5.958e-02 reward_prediction:7.399e-05 success_rate:8.100e-02
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> 20250326-15:00:17 trail_idx:020 total_steps:810 step:200 reward:0.000e+00 state_prediction:6.262e-02 reward_prediction:7.668e-05 success_rate:7.290e-02
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> Reach max_steps, break trail.
> Reach max_steps, break trail.

**Comment by SharingChen** [comment]
感谢回复,我还是想麻烦您更新一个可以与当前程序(generate.py)适配的示例文件, 包括配置文件和模型参数文件.
目前提供的两个配置文件,都存在一定问题,
config_16-128state_stage2_gsa_l32k_t512k_hot.yaml:   缺少很多配置参数,  rsa_type也与当前代码不适配了.
config.yaml: model_config的配置与当前提供的模型参数也不适配.
综上所述, 无法在不做修改的情况下 直接运行generate.py 得到预期实验结果, 主要还是因为配置文件中的配置项比较复杂,逐个探索会比较消耗时间.

**Comment by WorldEditors** [comment]
@SharingChen 非常抱歉，airsoul代码库我们近期迭代较快，包括修改了rsa_type的名称和位置参数等，老版本没有维护。可能需要您回滚下如下代码版本：https://github.com/airs-cuhk/airsoul/tree/42872debd327465b5222e8a1e667aac69f86aed3。 我们回头会TAG一个代码版本和对应的模型。

**Comment by SharingChen** [comment]
请问可以留一个方便沟通的邮箱吗，我在[Xenoverse]项目中提的关于AnyMDP的issue和PR无人回复，感谢。
